File: frc2018_5/src/main/java/ca/warp7/frc2018_5/Scottie.java
package ca.warp7.frc2018_5;

import ca.warp7.action.IAction;
import ca.warp7.action.impl.ActionMode;
import ca.warp7.frc.next.Robot;
import ca.warp7.frc2018_5.io.DriveOutput;
import edu.wpi.first.wpilibj.IterativeRobot;

public class Scottie extends IterativeRobot {

    @Override
    public void robotInit() {
        Robot.initSystems(
                DriveOutput.getInstance()
        );
    }

    @Override
    public final void disabledInit() {
        Robot.disable();
    }

    @Override
    public final void autonomousInit() {
        Robot.initAutonomousMode(new ActionMode() {
            @Override
            public IAction getAction() {
                return async();
            }
        }, 15);
    }

    @Override
    public final void teleopInit() {
        Robot.initTeleop(new TeleopRemote());
    }

    @Override
    public final void testInit() {
        Robot.initTest(null);
    }
}

File: frc2018_5/src/main/java/ca/warp7/frc2018_5/Scottie.kt
package ca.warp7.frc2018_5

import ca.warp7.frc2018_5.auto.Baseline
import ca.warp7.frc2018_5.input.DriveInput
import ca.warp7.frc2018_5.input.LiftInput
import ca.warp7.frc2018_5.input.NavX
import ca.warp7.frc2018_5.input.PoseEstimator
import ca.warp7.frc2018_5.output.*
import ca.warp7.frckt.*
import edu.wpi.first.wpilibj.IterativeRobot

class Scottie : IterativeRobot() {

    override fun robotInit() {
        println("Hello me is robit!")
        initRobotSystems(
                loopsPerSecond = 50,
                inputSystems = arrayOf(
                        DriveInput,
                        LiftInput,
                        NavX,
                        PoseEstimator
                ),
                outputSystems = arrayOf(
                        DriveOutput,
                        Superstructure,
                        LiftOutput,
                        ClimberOutput,
                        IntakeOutput,
                        WristOutput
                )
        )
    }

    override fun autonomousInit() = runRobotAuto(mode = Baseline, timeout = 15.0)
    override fun teleopInit() = startRobotTeleop(controlLoop = Controller)
    override fun testInit() = startRobotTest(controlLoop = Controller)
    override fun disabledInit() = disableRobot()
    override fun robotPeriodic() = Unit
    override fun autonomousPeriodic() = Unit
    override fun teleopPeriodic() = Unit
    override fun disabledPeriodic() = Unit
    override fun testPeriodic() = Unit
}